tts.py

```diff
@@ -120,23 +138,21 @@ async def ping(ctx):
 @bot.command()
 @commands.guild_only()
 async def setvoice(ctx, v="default"):
+    global role_cache
     v = v.capitalize()
-    role_list = list(map(lambda nm : get(ctx.guild.roles, name=nm), name_list))
-    #if the voice you want, v, is in the role list, then assign that.
-    if get(ctx.guild.roles, name=v) in role_list:
-        for r in role_list: # find any voice role user already has (in the role list) and remove it.
-            if r in ctx.author.roles:
-                await ctx.author.remove_roles(r) # removed.
-                break # should only have one of these so no need to continue this loop
-        await ctx.author.add_roles(get(ctx.guild.roles, name=v)) # assign new role.
+
+    # is `v` a valid voice role?
+    if get(role_cache, name=v) is not None:
+        # does user have any previous voice roles? (should only ever be one...)
+        prev = list(set(role_cache) & set(ctx.author.roles))
+        if len(prev) > 0:
+            # if so, remove it
+            await ctx.author.remove_roles(*prev)
+        # add the new voice role
+        await ctx.author.add_roles(get(role_cache, name=v))
     else:
-        chat = get(bot.get_all_channels(), name="chat")
-        chatmsg = 'Use the command `!setvoice voicename` to set a voice. Valid voicenames are '
-        for n in name_list:
-            if (n == 'Raul') or (n == 'Sabina'):
-                continue
-            chatmsg += '`' + n + '`, '
-        await chat.send(chatmsg)
+        # if not, just set the usage
+        await ctx.send(setvoice_help)

```

Comment:
if the previous channel selection was required, that can be added back in. for my testing server i only have one channel (not named chat)